src/v/utils/fragmented_vector.h

```diff
@@ -81,6 +81,12 @@ class fragmented_vector {
         return frag.at(index % elems_per_frag);
     }
 
+    T& operator[](size_t index) {
+        vassert(index < _size, "Index out of range {}/{}", index, _size);
+        auto& frag = _frags.at(index / elems_per_frag);
+        return frag.at(index % elems_per_frag);
+    }
```

Comment:
is this used in this PR?

Reply:
Yes, this is used in segment_index when we rewrite time values that were written for non-data batches.